game-core/src/main/java/games/strategy/engine/framework/map/file/system/loader/ZippedMapsExtractor.java

```diff
@@ -0,0 +1,170 @@
+      }
+
+      try {
+        Files.move(file.toPath(), extractionTarget.resolve(file.getName()));
+      } catch (final FileAlreadyExistsException e) {
+        log.error(
+            "Error, destination file already exists, failed to overwrite while unzipping map. Map: "
+                + mapZip.getAbsolutePath()
+                + ",file to write "
+                + extractionTargetFile.getAbsolutePath(),
+            e);
+        return;
+      }
+    }
+  }
+
+  /** Find .properties suffixed map files and renames them to match the output map file. */
+  private static void renameZipPropertiesFile(final File mapZip) {
+    final String newName = mapZip.getName().replace(".zip", "") + ".properties";
+
+    final String oldPropertiesFileName = mapZip.getName() + ".properties";
+    final Path oldPropertiesFilePath = mapZip.toPath().getParent().resolve(oldPropertiesFileName);
+    if (oldPropertiesFilePath.toFile().exists()) {
+      final Path newFilePath = mapZip.toPath().getParent().resolve(newName);
+      try {
+        log.info("Renaming {} -> {}", oldPropertiesFilePath, newFilePath);
+        Files.move(oldPropertiesFilePath, newFilePath);
+      } catch (final IOException e) {
+        throw new FileSystemException(
+            "Failed to rename file: " + oldPropertiesFilePath + " to " + newFilePath, e);
+      }
+    }
+  }
+
+  private static void removeMapZip(final File mapZip) {
+    log.info("Removing map zip: {}", mapZip.getAbsolutePath());
+    final boolean removed = mapZip.delete();
+    if (!removed) {
+      log.info(
+          "Failed to remove (extracted) zip file: {}, marking file to be deleted on exit.",
+          mapZip.getAbsolutePath());
+      mapZip.deleteOnExit();
+    }
+  }
+
+  /**
+   * Moves a target zip file into a 'bad-zip' folder. This is to prevent the file from being picked
+   * up in future unzip operations and cause repeated warning messages to users.
+   *
+   * @return Returns the new location of the file, returns an empty if the file move operation
+   *     failed.
+   */
+  private static Optional<Path> moveBadZip(final File mapZip) {
+    final Path badZipFolder =
+        ClientFileSystemHelper.getUserMapsFolder().toPath().resolve("bad-zips");
+    if (!badZipFolder.toFile().mkdirs()) {
+      log.error(
+          "Unable to create folder: "
+              + badZipFolder.toFile().getAbsolutePath()
+              + ", please report this to TripleA and create the folder manually.");
+      return Optional.empty();
+    }
+    try {
+      final Path newLocation = badZipFolder.resolve(mapZip.getName());
+      Files.move(mapZip.toPath(), newLocation);
+
+      return Optional.of(newLocation);
+    } catch (final IOException e) {
+      log.error(
+          "Failed to move file: "
+              + mapZip.getAbsolutePath()
+              + ", to: "
+              + badZipFolder.toFile().getAbsolutePath(),
+          e);
+      return Optional.empty();
+    }
+  }
+}
```

Comment:
Ah one thing I forgot to mention:
The Zipped Maps Extractor is just a migration tool right?
If so, would this be a candidate for a class that should be removed in the next major release or some other point in time?